std/js/lib/BigInt.hx

```diff
@@ -0,0 +1,223 @@
+		return if (locales == null) {
+			if (options == null) {
+				this.toLocaleString();
+			} else {
+				this.toLocaleString(js.Lib.undefined, options);
+			}
+		} else {
+			if (options == null) {
+				this.toLocaleString(locales);
+			} else {
+				this.toLocaleString(locales, options);
+			}
+		}
+	}
+
+	/**
+		Returns a string representing the specified object in the specified radix (base).
+		Overrides the `Object.prototype.toString()` method.
+	**/
+	public inline function toString(?radix:Int):String {
+		return if (radix == null) {
+			this.toString();
+		} else {
+			this.toString(radix);
+		}
+	}
+
+	/**
+		Returns the primitive value of the specified object.
+		Overrides the `Object.prototype.valueOf()` method.
+	**/
+	public inline function valueOf():BigInt {
+		return this.valueOf();
+	}
+
+	/**
+		Wraps a BigInt value to a signed integer between `-2^(width-1)` and `2^(width-1)-1`.
+	**/
+	public static inline function asIntN(width:Int, bigint:BigInt):BigInt {
+		return _BigInt.asIntN(width, bigint);
+	}
+
+	/**
+		Wraps a BigInt value to an unsigned integer between `0` and `2^width-1`.
+	**/
+	public static inline function asUintN(width:Int, bigint:BigInt):BigInt {
+		return _BigInt.asUintN(width, bigint);
+	}
+
+	@:op(-A) static inline function neg(a:BigInt):BigInt {
+		return (Syntax.code("(-{0})", a));
+	}
+
+	@:op(++A) static inline function incPre(a:BigInt):BigInt {
+		return Syntax.code("(++{0})", a);
+	}
+
+	@:op(A++) static inline function incPost(a:BigInt):BigInt {
+		return Syntax.code("({0}++)", a);
+	}
+
+	@:op(--A) static inline function decPre(a:BigInt):BigInt {
+		return Syntax.code("(--{0})", a);
+	}
+
+	@:op(A--) static inline function decPost(a:BigInt):BigInt {
+		return Syntax.code("({0}--)", a);
+	}
+
+	@:op(A + B) static inline function add(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} + {1})", a, b);
+	}
+
+	@:op(A - B) static inline function sub(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} - {1})", a, b);
+	}
+
+	@:op(A * B) static inline function mul(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} * {1})", a, b);
+	}
+
+	@:op(A / B) static inline function div(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} / {1})", a, b);
+	}
+
+	@:op(A % B) static inline function mod(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} % {1})", a, b);
+	}
+
+	@:op(~A) static inline function not(a:BigInt):BigInt {
+		return Syntax.code("~{0}", a);
+	}
+
+	@:op(A & B) static inline function and(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} & {1})", a, b);
+	}
+
+	@:op(A | B) static inline function or(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} | {1})", a, b);
+	}
+
+	@:op(A ^ B) static inline function xor(a:BigInt, b:BigInt):BigInt {
+		return Syntax.code("({0} ^ {1})", a, b);
+	}
+
+	@:op(A << B) static inline function lshift(a:BigInt, b:BigInt):BigInt {
```

Comment:
Are you sure bit shift ops require a `BigInt` as the right hand argument?

Reply:
Yes. In V8(Chrome and Node.js),

```
> 1n << 1n
Uncaught TypeError: Cannot mix BigInt and other types, use explicit conversions
```

But Firefox accepts int.

Reply:
Oh, I find a mistake of sample code... (This is already fixed.)

It works fine.

```js
> 1n << 1n
2n
```

It doesn't work.

```js
> 1n << 1
Thrown:
TypeError: Cannot mix BigInt and other types, use explicit conversions
```